tools/regenerate-docs.ts

```diff
@@ -126,8 +160,11 @@ details.forEach(({ name, description }) => {
 
   const contents = fs.readFileSync(pathToDoc).toString().split('\n');
 
+  // Replace the title.
   contents[0] = `# ${description} (\`${name}\`)`;
 
+  updateRuleNotices(contents, name);
```

Comment:
Relating to my comment above since we're always working with the very start of the file, I think we should replace this with `removeHeader` + `generateHeader` functions, and then the `generateHeader` function can be simplified by just being an array:

```
const generateHeader = () => {
  const header = [`# ${description} (\`${name}\`)`];

  if (/* rule is in a config */) {
    header.push('', '$ this rule is in configs ...');
  }

  // etc

  return header.join('\n');
}
```

Reply:
I went ahead and refactored to something like this -- `generateRuleHeaderLines()` + `replaceOrCreateHeader()`